main.py

```diff
@@ -481,6 +486,94 @@ def delete_interface_metadata(self, request: Request) -> JSONResponse:
         self.notify_metadata_changes(interface, 'removed')
         return JSONResponse("Operation successful")
 
+    @staticmethod
+    def map_singular_values(tag_range):
+        """Change integer or singular interger list to
+        list[int, int] when necessary"""
+        if isinstance(tag_range, int):
+            tag_range = [tag_range] * 2
+        elif len(tag_range) == 1:
+            tag_range = [tag_range[0]] * 2
+        return tag_range
+
+    def _get_tag_ranges(self, content: dict):
+        """Get tag_ranges and check validity:
+        - It should be ordered
+        - Not unnecessary partition (eg. [[10,20],[20,30]])
+        - Singular intergers are changed to ranges (eg. [10] to [[10, 10]])
+        The ranges are understood as [inclusive, inclusive]"""
+        ranges = content["tag_ranges"]
+        if len(ranges) < 1:
+            detail = "tag_ranges is empty"
+            raise HTTPException(400, detail=detail)
+        last_tag = 0
+        for i, _ in enumerate(ranges):
```

Comment:
If you've changed your mind here and only used and index then the `enumerate` could be just a `range`.

Reply:
Yeah the linter recommended this change, changed back to `range`